testsuite/src/main/scala/endpoints/testsuite/BasicAuthTestApi.scala

```diff
@@ -12,4 +13,13 @@ trait BasicAuthTestApi extends algebra.Endpoints with algebra.BasicAuthenticatio
     textResponse
   )
 
+  val protectedEndpointViaBuilder =
+    anEndpoint
+      .withMethod(Get)
+      .withUrl(path / "users")
+      .withTextResponse
+      .withBasicAuth
```

Comment:
@julienrf I am confused by your statement about auth happening only for requests, not responses.  Could you show an example of the non-builder syntax of the same bit of code?
Also, isn't authentication only for requests?  I don't think I have heard of an authenticated response.